src/compiler/checker.ts

```diff
@@ -16725,7 +16725,13 @@ namespace ts {
                 // only the class declaration node will have the Abstract flag set.
                 const valueDecl = expressionType.symbol && getClassLikeDeclarationOfSymbol(expressionType.symbol);
                 if (valueDecl && hasModifier(valueDecl, ModifierFlags.Abstract)) {
-                    error(node, Diagnostics.Cannot_create_an_instance_of_the_abstract_class_0, declarationNameToString(getNameOfDeclaration(valueDecl)));
+                    const name = getNameOfDeclaration(valueDecl);
+                    if (name) {
+                        error(node, Diagnostics.Cannot_create_an_instance_of_the_abstract_class_0, declarationNameToString(name));

```

Comment:
Would it make sense to get the name from the import alias or local variable (for class expressions) if the class doesn't have a name?